### PR TITLE
Build and deploy DuckDB-Wasm extensions

### PR DESCRIPTION
**Pull request by carlopi**
NOTE: Last commit has to be reverted, moving to main once https://github.com/duckdb/duckdb/pull/9356 is merged

Build step is tested, signing step is tested (but for verifying the binaries are actually valid, need to add that), deploy step untested.

DuckDB side PR has been merged, bumped to main the workflow.

### CODE REVIEW

**Comment by carlopi** [comment]
I did ended up testing also the deploy, once it's green here it should be good to go.

File: .github/workflows/_extension_deploy_wasm.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
+#
+# Reusable workflow that deploys the artifacts produced by github.com/duckdb/duckdb/.github/workflows/_extension_distribution.yml
```
Comment (samansmink) [inline .github/workflows/_extension_deploy_wasm.yml:2]: Is there a reason this is a separate workflow? Shouldn't this just be added to the already existing `_extension_deploy.yml` ?

**Review by samansmink — COMMENTED** [review]
Hey @carlopi looks good! very excited to have this working. I added two small comments!

File: .github/workflows/_extension_deploy_wasm.yml (hunk 1 above)
Comment (carlopi) [reply .github/workflows/_extension_deploy_wasm.yml:2]: I will merge the two, makes sense and it's cleaner.

**Comment by samansmink** [comment]
Good stuff!

**Pull request merged**